app/util/analytics.py

```diff
@@ -83,7 +101,9 @@ def record_event(tag, log_msg=None, **event_data):
         event_data['__id__'] = _event_id_generator.increment()
         event_data['__tag__'] = tag
         event_data['__timestamp__'] = time.time()
-        _analytics_logger.event(json.dumps(event_data, sort_keys=True))  # pylint: disable=no-member
+        json_dumps = json.dumps(event_data, sort_keys=True)
+        _analytics_logger.event(json_dumps)  # pylint: disable=no-member
+        write_to_log_cache(json_dumps)
```

Comment:
Should we cache the un-jsonified event so we don't need to `json.loads` it again?